modules/platforms/cpp/core/src/impl/cluster/cluster_group_impl.cpp

```diff
@@ -231,6 +355,29 @@ namespace ignite
                 return FromTarget(res);
             }
 
+            SP_ClusterGroupImpl ClusterGroupImpl::ForPredicate(IgnitePredicate<ClusterNode>* pred)
+            {
+                SP_PredicateHolder newPredHolder(new ClusterNodePredicateHolder());
+
+                newPredHolder.Get()->Insert(pred);
+                newPredHolder.Get()->Insert(*predHolder.Get());
+
+                std::vector<Guid> nodeIds;
+                std::vector<ClusterNode> allNodes = GetNodes();
+                for (size_t i = 0; i < allNodes.size(); i++)
+                    if (newPredHolder.Get()->operator()(allNodes.at(i)))
+                        nodeIds.push_back(allNodes.at(i).GetId());
+
+                if (nodeIds.empty())
+                    throw IgniteError(IgniteError::IGNITE_ERR_CLUSTER_GROUP_EMPTY, "The empty cluster group cannot be created.");
```

Comment:
As, you can see, in case of ForPredicate(), the new cluster group is created using ForNodeIds()  call. And if I will pass empty container of nodes here it will throw exception like "Argument is invalid: ids must not be empty."

Now I changed the behavior and the empty cluster group will be returned.
Please check, that empty cluster group is created correctly in this case.